jupyterhub_singleuser_profiles/profiles.py

```diff
@@ -101,21 +120,62 @@ def write_config_map(self, config_map_name, key_name, data):
       else:
         raise
 
+  def write_secret(self, secret_name, key_name, data):
+    secret = V1Secret()
+    secret.metadata = V1ObjectMeta(name=secret_name, labels={'app': 'jupyterhub'})
+    secret.string_data = {key_name: yaml.dump(data, default_flow_style=False)} #stringData instead of data here to make kubernetes parse this as a string without base64 encoding
+    try:
+      api_response = self.api_client.replace_namespaced_secret(secret_name, self.namespace, secret)
+    except ApiException as e:
+      if e.status == 404:
+        try:
+          api_response = self.api_client.create_namespaced_secret(self.namespace, secret)
+        except ApiException as e:
+          _LOGGER.error("Exception when calling CoreV1Api->create_namespaced_secret: %s\n" % e)
+      else:
+        raise
+
   def update_user_profile_cm(self, username, data={}):
     
     user_cm = self.get_user_profile_cm(username)
+    user_cm_env = []
+    user_secret = self.get_user_secret(username)
+    user_secret['env'] = []
```

Comment:
Is there a reason why you are handling CM and Secret differently - i.e.

For CM you create a separate variable and `append` to it, then set it to `cm['env']`
For Sercet you `append` directly to the `secret['env']`

Reply:
I am reading `user_cm['env']` and dividing it between what stays in `user_cm` and what goes into secrets. That's why I needed the extra variable.